rust/src/index/vector/pq.rs

```diff
@@ -392,7 +392,18 @@ impl ProductQuantizer {
         }
         let pd_centroids = codebook_builder.finish();
         self.codebook = Some(Arc::new(pd_centroids));
-        self.transform(&sub_vectors, metric_type).await
+
+        Ok(())
+    }
+
+    /// Train a [ProductQuantizer] using an array of vectors.
+    pub async fn fit_transform(
+        &mut self,
+        mat: &MatrixView,
+        metric_type: MetricType,
+    ) -> Result<FixedSizeListArray> {
+        self.train(mat, metric_type, 50).await?;
```

Comment:
Should we move hard coded constants into one place so we know what needs to change in the future?

Reply:
oh, yea, lemme just make this a parameter passed from user-facing api